### PR TITLE
Create a script to remove old data from the DB

### LINKED ISSUE #1957: Prepare script for database cleanup
Create a prototype script to delete older data - figure out how to [do that in SQLAlchemy](https://www.tutorialspoint.com/sqlalchemy/sqlalchemy_core_multiple_table_deletes.htm).
It'd probably need to be done with several statements in one transaction (for each pipeline) because we have [several interconnected tables](https://github.com/packit/packit-service/blob/main/CONTRIBUTING.md). Doing it with one statement/query can be doable but bad for performance and resource consumption. There might be data not assigned to any pipeline (from the pre-pipelines era) - delete those as well.

### :warning: warnings: 

* [ ] Do not test/run the script on deployed databases (prod/stg) yet.
* [ ] [Dump the data from stage and prod](https://github.com/packit/packit-service/blob/main/CONTRIBUTING.md#using-live-data-locally), and play with the script locally.
* [ ] Let the script be reviewed.

As a follow-up, we will automate the backup and cleanup: https://github.com/packit/packit-service/issues/1923

### PR DESCRIPTION
**Pull request by csomh**
Getting rid of old data speeds up API queries.

To find out how much, I've run 2 tests:
1. Measure how much time it takes to query `/api/runs` and iterating through 100 pages, with 10 results per page.
2. Measure how much time it takes to query `/api/copr-builds` and iterating through 100 pages, with 50 results per page.

| | `/api/runs` | `/api/copr-builds` |
| --- | --: | --: |
| before cleanup | 0:01:14.973 | 0:02:02.964 |
| keeping 1 year | 0:00:51.586 | 0:01:29.547 |
| % | -31.19% | -27.18% |
| keeping 6 months | 0:00:39.452 | 0:01:04.199 |
| % | -47.38% | -47.79% |

Times are  `hours:minutes:seconds.milliseconds`.

Resolves #1957.

### CODE REVIEW

**Review by jpopelka — APPROVED** [review]

**Review by lbarcziova — APPROVED** [review]
:+1:

**Comment by jpopelka** [comment]
Are there some "lessons learned the hard way" from your story about deleting the data from the database?
Something the team should have in mind if they ever want to update the script?

For example, I, after a bit of study, understand the simple `select().outerjoin().filter()`, but I'm screwed when an additional `.outerjoin_from()` comes to the party.
Is it explainable in layman's terms? :)

**Comment by csomh** [comment]
Lesson 1: understand the **whole** schema of the DB before you start deleting things, and identify the leaves. That's where deletion should start.

Lesson 2: it is difficult to delete data from complicated DBs, b/c of the intertwined relationships. Don't allow your DB to become more complicated than your problem domain.

Lesson 3: SQLAlchemy docs are complicated and are lacking clear examples. ChatGPT can give (mostly correct) usable examples, which helps understanding. 

Maybe more.

As for `.outerjoin_from()`: I used that, b/c the field selected is on the right side of the join. When doing `select().outerjoin()`, the argument of `select()` will end up on the left side, while the argument of `outerjoin()` will end up on the right side. `outerjoin_from()` takes both tables as an argument, and it's possible to use a `select()` in front of it, to pick a field from the second table.
